src/components/Image/src/Image.vue

```diff
@@ -145,54 +160,29 @@ export default {
 		if (this.loaded) {
 			this.$emit('image:visible');
 		}
-		if (!this.lazyload) {
-			this.load();
-		} else {
-			if (!observer) {
-				observer = new SharedIntersectionObserver();
-			}
+
+		if (this.lazyload) {
+			observer ??= new SharedIntersectionObserver();
+
 			observer.watch(this.$el, ({ isIntersecting }) => {
 				if (isIntersecting) {
 					this.load();
 				}
 			});
+		} else {
+			this.load();
 		}
+
 		this.getImageDimensions();
 	},
 
 	beforeDestroy() {
-		if (observer) {
-			observer.unwatch(this.$el);
-		}
+		observer?.unwatch(this.$el);
 	},
 
 	methods: {
 		load() {
-			if (this.loaded || (!this.src && !this.srcset)) {
-				return;
-			}
-
-			const img = new Image();
-
-			if (this.src) {
-				img.src = this.src;
-			}
-
-			if (this.srcset) {
-				img.srcset = this.srcset;
-			}
-
-			// Needed to not load the full size image
-			// and match the size loaded in the UI
-			if (this.sizes) {
-				img.sizes = this.sizes;
-			}
-
-			img.addEventListener('load', () => {
-				imgCache.add(this.src + this.srcset);
-				this.loaded = true;
-				this.getImageDimensions();
-			});
+			this.shouldLoad = true;
```

Comment:
Should this value be reset to `false` after the initial load?

Reply:
I don't think `shouldLoad` should ever change after it's been set to `true`. Changing it to `false` would cause the `src` and `srcset` values on the element to become unset, causing the image to unload.